probe-rs/src/bin/probe-rs/main.rs

```diff
@@ -92,22 +93,22 @@ pub(crate) struct CoreOptions {
 }
 
 /// A helper function to deserialize a default [`Format`] from a string.
-fn format_from_str<'de, D: Deserializer<'de>>(deserializer: D) -> Result<Format, D::Error> {
+fn format_from_str<'de, D: Deserializer<'de>>(deserializer: D) -> Result<Option<Format>, D::Error> {
```

Comment:
I think the format should directly have a default instead of an option.

Reply:
The problem is that you need to know if the argument was supplied or not, if it wasn't supplied then we look at the target definition. Any ideas on how we can do that with clap?

Reply:
Ah, silly me, that's indeed tricky. clap has some conditional flags but i dont think for this.